### ISSUE TITLE
[HOLD for payment 2023-06-13] [$1000] Dev: Not able to run storybook

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#19777]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Navigate to the app folder on terminal
2. Run command: `npm run storybook`

## Expected Result:
Not able to run storybook

## Actual Result:
Crash during building storybook

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** n/a

**Reproducible in staging?:** n/a

**Reproducible in production?:** n/a

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

![image (45)](https://github.com/Expensify/App/assets/93399543/e1d3910e-5c05-41be-9e25-5af278f7cd19)

**Expensify/Expensify Issue URL:**

**Issue reported by:** Robert Kozik (@robertKozik) 

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1685043545129419

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01ba9220f901f5a917</li>
        <li>Upwork Job ID: 1664323419533189120</li>
        <li>Last Price Increase: 2023-06-01</li>
    </ul>
</details>

**Comment by sakluger** [comment]
I don't have a dev environment, but @neil-marcellini was able to reproduce. Marking as external.

**Comment by sakluger** [comment]
Oh wait, this is a duplicate of https://github.com/Expensify/App/issues/15042. Closing.

**Comment by neil-marcellini** [comment]
Oh yeah this is different from #15042, where the problem is opening storybook on staging.

**Comment by s77rt** [comment]
In case we wanted to fix this silently, I have drafted a [PR](https://github.com/Expensify/App/pull/19980) already. Basically this fails because the env file path passed from storybook config is `../.env.production` and we split the path as is on the dots taking the 3rd element in this case it will be `/` which is not a valid environment. The solution is extract the filename from the path and apply same operation on the filename (not the whole path)..

**Comment by sakluger** [comment]
Sorry about that, thanks for reopening! I've added the external label, looks like we already have a few proposals which is great.

**Comment by neil-marcellini** [comment]
@francoisl is actually managing this one, but I'd be happy to take it over if you want Francois.

**Comment by francoisl** [comment]
Looks good to me, let's go with your proposal @allroundexperts!

**Comment by sakluger** [comment]
I'm heading OOO for the next week so I'm going to assign to another BZ member. I went ahead and sent offers for @allroundexperts and @fedirjh.

@robertKozik can you link your Upwork profile so that we can send you an offer for the bug report?

**Comment by jliexpensify** [comment]
Hi @sakluger you want to take this one as B0? I notice you re-assigned yourself.

**Comment by sakluger** [comment]
@jliexpensify when we are headed OOO, we're supposed to assign another team member but also leave ourself assigned according to this Slack post: https://expensify.slack.com/archives/C01SKUP7QR0/p1684185896898599?thread_ts=1684163684.524419&cid=C01SKUP7QR0

**Comment by jliexpensify** [comment]
Hmm I see your OOO comment now, but it weirdly didn't show up when I glanced at this issue the other day. Sorry about that!

**Comment by jliexpensify** [comment]
Hi, I can pay this if Sasha is OOO. Looks like it was merged within a day so there's a bonus applicable.

@francoisl - is there any need to fill out the [checklist](https://github.com/Expensify/App/issues/19777#issuecomment-1588266720)?

**Comment by francoisl** [comment]
Yeah, since we added automated tests, I agree we don't need to add other regression tests.

Maybe let's at least identify the PR that added the regression and make sure that it was tested properly? So basically do the first two checkboxes of the checklist.

**Comment by jliexpensify** [comment]
@robertKozik I can't seem to find you on Upworks, can you apply here for the $250 reporting bonus? 

https://www.upwork.com/jobs/~01ba9220f901f5a917

**Comment by jliexpensify** [comment]
@fedirjh and @allroundexperts - have paid you with bonus. Waiting on @robertKozik

**Comment by robertKozik** [comment]
Hi! Im contributor from Software Mansion agency so I'm not eligible to payment 😄

**Comment by jliexpensify** [comment]
Yeah that'd explain it, thanks Robert!

**Comment by jliexpensify** [comment]
Job removed, closing!
